xbmc/network/WakeOnAccess.cpp

```diff
@@ -68,6 +73,117 @@ static unsigned long HostToIP(const std::string& host)
   return inet_addr(ip.c_str());
 }
 
+#define LOCALIZED(id) g_localizeStrings.Get(id)
+
+static void ShowDiscoveryMessage(const char* function, const char* server_name, bool new_entry)
+{
+  std::string message, heading = LOCALIZED(13033);
+
+  if (new_entry)
+  {
+    CLog::Log(LOGDEBUG, "%s - Create new entry for host '%s'", function, server_name);
+    message = StringUtils::Format(LOCALIZED(13035).c_str(), server_name);
+  }
+  else
+  {
+    CLog::Log(LOGDEBUG, "%s - Update existing entry for host '%s'", function, server_name);
+    message = StringUtils::Format(LOCALIZED(13034).c_str(), server_name);
+  }
+  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, heading, message, 4000, true, 3000);
+}
+
+struct UPnPServer
+{
+  UPnPServer()
+  {
+    nextWake = CDateTime::GetCurrentDateTime();
+  }
+  bool operator == (const UPnPServer& server) const { return server.uuid == uuid; }
+  bool operator == (const std::string& server_uuid) const { return server_uuid == uuid; }
+  std::string name;
+  std::string uuid;
+  std::string mac;
+  CDateTime nextWake;
+};
+
+typedef std::vector<UPnPServer> upnp_server_list_t;
+
+static upnp_server_list_t upnp_servers;
+
+static UPnPServer* LookupUPnPServer(const std::string& uuid)
+{
+  upnp_server_list_t::iterator server_it = find(upnp_servers.begin(), upnp_servers.end(), uuid);
+
+  return server_it != upnp_servers.end() ? &(*server_it) : 0;
```

Comment:
This need to be changed, whenever the vector grows this will point at random memory
Use a shared_ptr instead or possibly a reference.
Also, use nullptr for invalid pointers and never ever 0